module/VuFind/tests/integration-tests/src/VuFindTest/Mink/SavedSearchesTest.php

```diff
@@ -78,7 +78,18 @@ public function setUp(): void
     public function testSaveSearch()
     {
         $page = $this->performSearch('test');
-        $this->clickCss($page, '.fa.fa-save');
+        $links = $page->findAll('css', '.searchtools a');
+        $saveLink = null;
+        foreach ($links as $link) {
+            if ($this->checkVisibility($link)
+                && str_contains($link->getHtml(), 'Save Search')
+            ) {
+                $saveLink = $link;
+                break;
+            }
+        }
+        $this->assertNotNull($link);
```

Comment:
I think you want to be asserting/clicking $saveLink, not $link.

Reply:
Yes! It looks like that `break` has been keeping this running correctly.